stable_baselines/common/base_class.py

```diff
@@ -150,6 +153,89 @@ def _setup_learn(self, seed):
         if seed is not None:
             set_global_seeds(seed)
 
+    @abstractmethod
+    def _get_pretrain_placeholders(self):
+        """
+        Return the placeholders needed for the pretraining:
+        - obs_ph: observation placeholder
+        - actions_ph will be population with an action from the environement
+            (from the expert dataset)
+        - deterministic_actions_ph: e.g., in the case of a gaussian policy,
+            the mean.
+
+        :return: ((tf.placeholder)) (obs_ph, actions_ph, deterministic_actions_ph)
+        """
+        pass
+
+    def pretrain(self, dataset, num_iter=1000, batch_size=64,
+                 learning_rate=1e-4, adam_epsilon=1e-8):
+        """
+        Pretrain a model using behavior cloning:
+        supervised learning given an expert dataset.
+
+        NOTE: only Box and Discrete spaces are supported for now,
+        and images are not supported properly.
+
+        :param dataset: (Dataset) Dataset manager
+        :param num_iter: (int) Number of iterations (gradient steps)
+        :param batch_size: (int) minibatch size
+        :param learning_rate: (float) Learning rate
+        :param adam_epsilon: (float) the epsilon value for the adam optimizer
+        :return: (BaseRLModel) the pretrained model
+        """
+        continuous_actions = isinstance(self.action_space, gym.spaces.Box)
+        discrete_actions = isinstance(self.action_space, gym.spaces.Discrete)
+
+        assert discrete_actions or continuous_actions, 'Only Discrete and Box action spaces are supported'
+
+        # Validate the model every 10% of the total number of iteration
+        val_interval = int(num_iter / 10)
+
+        with self.graph.as_default():
+            if continuous_actions:
```

Comment:
shouln't all the pretraining operation be in the `with tf.variable_scope('pretrain'):` block?

Reply:
good point